lib/promscrape/scrapework.go

```diff
@@ -344,8 +344,8 @@ func (sw *scrapeWork) run(stopCh <-chan struct{}, globalStopCh <-chan struct{})
 				sw.seriesLimiter = nil
 			}
 			return
-		case tt := <-ticker.C:
-			t := tt.UnixNano() / 1e6
+		case <-ticker.C:
+			t := time.Now().UnixNano() / 1e6
 			if d := math.Abs(float64(t - timestamp)); d > 0 && d/float64(scrapeInterval.Milliseconds()) > 0.1 {
 				// Too big jitter. Adjust timestamp
 				timestamp = t
```

Comment:
previously scrapeTimestamp will increase with factor of scrapeInterval. e.g. interval=5s:
[`t0`, `t0+5s`, `t0+10s`, (something happend and skipped scrape), `t0+20s`]

But if we use `time.Now()` here, it could be random:
[`t0`, `t0+5s`, `t0+10s`, (something happend and skipped scrape), `t0+17s`]

One possible way is to skip the scrape when the diff > 50% (TBD):
```go
		case tt := <-ticker.C:
			t := time.Now().UnixNano() / 1e6
			if d := math.Abs(float64(t - timestamp)); d > 0 && d/float64(scrapeInterval.Milliseconds()) > 0.5 {
				// Too big jitter. skip this scrape?
			}
```

Reply:
>previously scrapeTimestamp will increase with factor of scrapeInterval. e.g. interval=5s:
[t0, t0+5s, t0+10s, (something happend and skipped scrape), t0+20s]
But if we use time.Now() here, it could be random:
[t0, t0+5s, t0+10s, (something happend and skipped scrape), t0+17s]

Should the current scrapeTimestamp be `t0+15s` instead of `t0+20s`?
```
[t0, t0+5s, t0+10s, (something happend and skipped scrape), t0+15s]

if use time.Now():
[t0, t0+5s, t0+10s, (something happend and skipped scrape), t0+17s]
```

Reply:
I don’t think we need to skip any scrapes, the existing adjustment for 10% scrapeInterval jitter seems good to me.
This fix does provide more accurate states when having slow scrapes, such as for `vm_promscrape_scrape_duration_seconds` and target status.